python/cudf/cudf/core/column/column.py

```diff
@@ -916,13 +916,17 @@ def _obtain_isin_result(self, rhs: ColumnBase) -> ColumnBase:
         Helper function for `isin` which merges `self` & `rhs`
         to determine what values of `rhs` exist in `self`.
         """
-        ldf = cudf.DataFrame({"x": self, "orig_order": arange(len(self))})
-        rdf = cudf.DataFrame(
-            {"x": rhs, "bool": full(len(rhs), True, dtype="bool")}
-        )
-        res = ldf.merge(rdf, on="x", how="left").sort_values(by="orig_order")
-        res = res.drop_duplicates(subset="orig_order", ignore_index=True)
-        return res._data["bool"].fillna(False)
+        # We've already matched dtypes by now
+        result = libcudf.search.contains(rhs, self)
+        # libcudf contains runs the search with nulls comparing equal
+        # and then copying the bitmask from the needles to the result.
+        # In cudf, we want nulls in both needle and haystack to
+        # produce True and nulls in only one to produce False. If we
+        # drop the bitmask of the result on the floor, it already
+        # contains the values we want. We can do this unilaterally
+        # without checking the null count of the result.
+        result.set_base_mask(None)
```

Comment:
This happens to be correct because `cudf::contains` writes all the output and then adds a null mask. But that is not in the contract for a column with a null mask, the masked entries are allowed to contain anything. We can do it "above-board" by an appropriate `fillna` call here. Or we could augment the libcudf API to allow control of what happens when nulls are in the needles.

Reply:
I don't know if the libcudf API is being used by others elsewhere, but I'm okay having the fillna call here.

Reply:
Yeah, though it's another kernel call + memory allocation. I'll do that and open a feature req.

Reply:
https://github.com/rapidsai/cudf/issues/14515

Reply:
OK, did the above-board thing for now.